pkg/controller/accountclaim/accountclaim_controller_test.go

```diff
@@ -142,6 +136,57 @@ var _ = Describe("AccountClaim", func() {
 			Expect(err).To(HaveOccurred())
 			Expect(err.Error()).To(Equal("account CR modified during reset: Conflict"))
 		})
+
+		When("Accountclaim is BYOC", func() {
+
+			BeforeEach(func() {
+				accountClaim.SetFinalizers(append(accountClaim.GetFinalizers(), accountClaimFinalizer))
+				accountClaim.Spec.BYOC = true
+				accountClaim.Spec.AccountLink = ""
+			})
+
+			It("should fail validation", func() {
+				// this should cause Validate() to fail
```

Comment:
Did I get that right?

```suggestion
				// fail validation if BYOC is not associated with an account
```

Reply:
Yep, that's better!